src/main/java/org/candlepin/model/ProductContent.java

```diff
@@ -44,6 +48,32 @@
 @Immutable
 @Cache(usage = CacheConcurrencyStrategy.READ_ONLY)
 @Table(name = ProductContent.DB_TABLE)
+@SqlResultSetMapping(
```

Comment:
unlike the previous solution we use native query and it doesn't support mapping to object if there are other attributes in the result than from the given class so I had to use this functionality

Reply:
nack on this

This is a major failing on the part of the JPA spec in that to properly use this mapping, we have to define it with annotations, in a place the annotation scanner picks it up, which is likely in a separate location from the query that requires it. This is just extra maintenance pain for something that should absolutely not be defined with annotations, or at the very least, should be something that can be manually constructed with POJOs.

Additionally, due to the inconsistent behavior with the default mapping vs this mapping with regards to field strictness, it's not clear what would happen in the future if a column in the defined list were to go missing or get renamed. Also, will Content instances still lazy-load their collection fields when fetched using this mapping?

As a distant third, we're also using the splat operator in the native query itself, which isn't great for our production usage since we can no longer rely upon the order or presence of any given field.

It's an interesting solution overall, but JPA's interface for implementing this behavior is unacceptably terrible, and creates an unnecessary maintenance burden and code bloat.

Reply:
I was looking for recommended approach and found that. I also find it strange that you can actually define a mapping above the table definition and for unrelated tables and then use it anywhere.

Best practife for this example is to use something like this: `SELECT content.uuid as uuid` so if you rename attribute you dont need to change mapping and if you delete attribute it is ok it can handle that you would only have unused code in mapping. 

Yes, it will still be lazy-loaded.

But I agree with you that it's terrible. Definitely the mapping from the second command seems better or can you think of a better way?